### PR TITLE
feat(base-cluster/limitrange-quotas): add limitrange and resource quota

### PR DESCRIPTION
**Pull request by marvinWolff**
Add limitRanges and Quotas 

Ref.: https://github.com/teutonet/teutonet-helm-charts/issues/655

### CODE REVIEW

**Comment by cwrau** [comment]
Can you add a CI values yaml with some values set and not set and so on?

**Review by cwrau — CHANGES_REQUESTED** [review]
Can you add a CI values yaml with some values set and not set and so on?

File: charts/base-cluster/templates/global/namespaces.yaml [hunk 1]
```diff
@@ -8,4 +8,43 @@ metadata:
...
+metadata:
+  name: {{ $name }}
+  namespace: {{ $name }}
+  labels: {{- include "common.labels.standard" $ | nindent 4 }}
+spec:
+  {{- $limits := dict "type" "Container" "defaultRequest" (dict "cpu" "100m" "memory" "100Mi")  -}}
+  {{ if $namespace.resources }}
+    {{- if $namespace.resources.defaults -}}
+      {{/* this section defines default limits */}}
+      {{- with $namespace.resources.defaults.limits -}}
```
Comment (cwrau) [inline charts/base-cluster/templates/global/namespaces.yaml:22]: Could you make these fields templatable?

File: charts/base-cluster/templates/global/namespaces.yaml [hunk 2]
```diff
@@ -8,4 +8,43 @@ metadata:
...
+        {{- $limits = set $limits "default" . -}}
+      {{- end -}}
+      {{/* this section defines default requests */}}
+      {{- with $namespace.resources.defaults.requests -}}
```
Comment (cwrau) [inline charts/base-cluster/templates/global/namespaces.yaml:26]: And these as well

File: charts/base-cluster/templates/global/namespaces.yaml [hunk 3]
```diff
@@ -8,4 +8,43 @@ metadata:
...
+---
+apiVersion: v1
+kind: ResourceQuota
+metadata:
+  name: {{ $name }}
+  namespace: {{ $name }}
+  labels: {{- include "common.labels.standard" $ | nindent 4 }}
+spec:
+  {{- $quotas := dict -}}
+  {{- with $namespace.resources.quotas -}}
```
Comment (cwrau) [inline charts/base-cluster/templates/global/namespaces.yaml:43]: And these as well

File: charts/base-cluster/templates/global/namespaces.yaml [hunk 4]
```diff
@@ -8,4 +8,43 @@ metadata:
...
+  {{- $quotas = . }}
+  hard: {{- $quotas | toYaml | nindent 4 }}
```
Comment (cwrau) [inline charts/base-cluster/templates/global/namespaces.yaml:45]:
```suggestion
  {{- with $namespace.resources.quotas -}}
  hard: {{- . | toYaml | nindent 4 }}
```

File: charts/base-cluster/templates/global/namespaces.yaml [hunk 5]
```diff
@@ -8,4 +8,43 @@ metadata:
...
+{{ if $namespace.resources.quotas }}
+---
+apiVersion: v1
+kind: ResourceQuota
+metadata:
+  name: {{ $name }}
+  namespace: {{ $name }}
+  labels: {{- include "common.labels.standard" $ | nindent 4 }}
+spec:
+  {{- $quotas := dict -}}
```
Comment (cwrau) [inline charts/base-cluster/templates/global/namespaces.yaml:42]:
```suggestion
```

**Review by cwrau — CHANGES_REQUESTED** [review]

**Comment by marvinWolff** [comment]
> Can you add a CI values yaml with some values set and not set and so on?

This is done, isn't it?

File: charts/base-cluster/values.yaml
Comment (cwrau) [inline charts/base-cluster/values.yaml:1]: If you don't want to add resource defaults for other namespaces, this looks fine to me 👌

**Review by cwrau — APPROVED** [review]

**Pull request merged**